apartmenttherapy/oxygen-lib-ui
# Replace moment with date-fns

#### What does this PR do & why?
+ This PR is an attempt to replace moment in oxy-lib-ui. Hopefully nothing was overlooked but I would love some extra eyes on this.

#### How can this be manually tested?
+ `Yarn test` should pass 
+ Confirm teasers on homepage have correct time
+ Confirm there is no weirdness with liftigniter
+ Test it live here: https://replacemoment-dot-oxygen-kitchn-dot-gen-prod.appspot.com/

#### Any questions or concerns about this change?
+ Take a look at the following example (notice the time of the teaser "10 easy & classic easer recipes"):

__Moment__ (time: "An hour ago")
<img width="1399" alt="Screenshot 2019-04-18 15 23 07" src="https://user-images.githubusercontent.com/147572/56390341-70439280-61f1-11e9-8fbd-a674776dcca9.png">

__Date-FNS__ (time: "Just Now")
<img width="1369" alt="Screenshot 2019-04-18 15 23 18" src="https://user-images.githubusercontent.com/147572/56390424-9d904080-61f1-11e9-9242-eceb952f11dd.png">

Now the reason for this difference is between moment's `.diff` hourly comparison and date-fns' `differenceInHours`. In this particular case the time was approx. 37 minutes since the post had been published. In moment's case it will return that there is an hourly difference due to the fact it returns a float and rounds. But date-fns will simply compare the two dates and return a 0 or x hour differential. @andrewb is this correct? It seems odd to make posts seem older then they already are.

<img width="748" alt="Screenshot 2019-04-18 15 22 49" src="https://user-images.githubusercontent.com/147572/56390496-c7496780-61f1-11e9-85fc-9bcea60e0ec1.png">

You can check the breakdown of oxy-lib-ui by running `BUNDLE_ANALYZE=browser yarn build`

__origin/master__
<img width="735" alt="Screenshot 2019-04-18 15 44 37" src="https://user-images.githubusercontent.com/147572/56390922-d7ae1200-61f2-11e9-9c23-0e38cafb0d21.png">

__wolff/replace-moment__
<img width="555" alt="Screenshot 2019-04-18 15 43 45" src="https://user-images.githubusercontent.com/147572/56390962-eeecff80-61f2-11e9-9dff-ff80bff07644.png">


#### Link to story card and other documentation
https://app.asana.com/0/0/1114999684123473/f

#### GIF
![](https://media.giphy.com/media/3oz8xKaR836UJOYeOc/giphy.gif)

#### Have you remembered to...?
